### PR TITLE
Migrate to Symfony 6, part 2

### LINKED ISSUE #34006: Upgrade to Symfony 6.0
The idea is to upgrade the composer dependency and upgrade to symfony 6, it also implies all the framework dependencies of course And it may imply upgrading some other dependencies we have that wouldn't be compatible with SF6 and need to be upgraded as well.

The target version should be as high as possible depending on the latest version when this task is done.

We must prevent the usage of already deprecated methods/code as much as possible.

> [!NOTE]  
> This hunt on deprecated code can be done in separated PRs if the upgrade PR is already too big

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrate to Symfony 6. This include stable UI tests.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | yes
| How to test?      | CI and tests UI are green
| UI Tests          | Legacy: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/7602332174; symfony: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/7602341941
| Fixed issue or discussion?     | https://github.com/PrestaShop/PrestaShop/issues/34006
| Related PRs       | -
| Sponsor company   | -

Deprecated:

We kept a compatibility layer in the FrameworkBundleAdminController for functions (these functions are normally no longer available in symfony 6)

- get
- has
- getDoctrine

Importants BC break:

- translator service is no longer public, but we can retrieve the PrestaShopBundle\Translation\TranslatorInterface public alias
- security.csrf.token_manager is no longer public, but we can retrieve the Symfony\Component\Security\Csrf\CsrfTokenManagerInterface public alias
- session and session.flash_bag services no longer exist, use RequestStack or Request instead

### CODE REVIEW

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModuleControllerRegisterPass.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+        if (preg_match('/namespace\s+(.+?);/s', $contents, $matches)) {
+            $namespace = $matches[1];
+        }
+
+        if (preg_match('/class\s+(\w+)\s+extends\s+FrameworkBundleAdminController/', $contents, $matches)) {
+            $className = $matches[1];
+        }
+
+        return $namespace ? $namespace . '\\' . $className : null;
```
Comment (jolelievre) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModuleControllerRegisterPass.php]:
```suggestion
        return !empty($namespace) && !empty($className) ? $namespace . '\\' . $className : null;
```
You need both values no? And maybe the `class_exists` can also be in this method

File: src/PrestaShopBundle/DependencyInjection/Compiler/ModuleControllerRegisterPass.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        foreach ($container->getDefinitions() as $definition) {
+            if ($definition->getClass() === $className) {
+                return $definition;
+            }
+        }
+
+        return $container->register($className, $className);
+    }
+
+    private function getClassNameFromFile(string $filePath): ?string
```
Comment (jolelievre) [inline src/PrestaShopBundle/DependencyInjection/Compiler/ModuleControllerRegisterPass.php]:
```suggestion
    private function getFrameworkAdminControllerClassNameFromFile(string $filePath): ?string
```

File: tests/Integration/Utility/AdapterSecurityAdmin.php [hunk 3]
```diff
@@ -28,24 +28,13 @@
...
-use PrestaShop\PrestaShop\Adapter\LegacyContext;
 use Symfony\Component\HttpKernel\Event\RequestEvent;
-use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
-use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
-use Symfony\Component\Security\Core\User\UserProviderInterface;
 
 /**
  * Admin Middleware security
  */
 class AdapterSecurityAdmin
```
Comment (jolelievre) [inline tests/Integration/Utility/AdapterSecurityAdmin.php:41]: You should add a comment here to explain that this listener aims at disabling the behaviour from the `Admin` listener in test environment

**Review by jolelievre — DISMISSED** [review]
Thanks @M0rgan01

**Review by jolelievre — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Pull request merged**